addons/mail/models/mail_thread.py

```diff
@@ -2009,14 +2009,22 @@ def message_post(self, *,
                  )
             )
         attachment_ids = list(attachment_ids or [])
-        if partner_ids and not tools.is_list_of(partner_ids, int):
-            raise ValueError(
-                _('Posting a message should receive partners as a list of IDs (received %(pids)s)',
-                  pids=repr(partner_ids),
-                 )
-            )
         partner_ids = list(partner_ids or [])
 
+
+        # def makePartner(partner):
+        #     if isinstance(partner, str):
+        #         partner = self.env['res.partner'].create({ "name": partner, "email": partner })
+        #         return partner.id
+
+
+        # partner_ids = list(map(makePartner, partner_ids))
+
+        partner_emails = [email for email in partner_ids if isinstance(email, str)]
+        partner_ids = map(lambda x: x.id, partner_ids + self.env['res.partner'].create([
```

Comment:
It’s not correct to always make a partner from suggested recipient. Some of them are already linked to a partner, therefore it should not create a partner for them. Only suggested recipients without any partner should automatically create a partner

Reply:
it looks like `partner.find_or_create` or `partner._find_or_create_from_emails()` can be used